### PR TITLE
Add action globbing (Closes #17)

### LINKED ISSUE #17: Globbing action folders.
Before I start so we are all on the same page, an "action" is the same thing as a command you run inside chat.  The naming difference is because calling it "command" conflicts with CLI ("command" line interface), and I wanted to avoid confusion inside of the code.  (We can rename that if you guys want, not a big deal, worth opening a separate issue though)

In order to load the actions, you point to a folder, and all of them at the base are loaded.

``` js
"actions": "./path/to/scripts"
// Default:
"actions": "node_modules/cordlr/lib/actions"
```

I think it would be more valuable if we used globing and arrays:

``` js
"actions": [
  "./path/to/scripts/**.js",
  "cordlr-analytics",
  // ...
]
```

This way, you can load scripts from anywhere you want, and also integrate 3rd party modules.

This will be valuable for us, to modularize the code, keeping the core nice and tidy.  Also helpful for other people who want to use this bot, not limited to our hardcoded commands.

Thoughts?

### PR DESCRIPTION
**Pull request by jamen**
Lets you do:

``` js
{
  // ...
  "actions": [
    "cordlr/actions/*.js", // Included scripts from cordlr
    "./custom/*.js", // Custom scripts
    "cordlr-analytics" // 3rd party scripts
  ]
}
```

Also uses `minimst` for commands inside chat, so you can do things like:

``` js
$role --foo
```

(One line addition, so why not)

### CODE REVIEW

**Comment by jamen** [comment]
Also closes #24

**Review by bdbch — APPROVED** [review]
looks good for me!

**Comment by bdbch** [comment]
@jamen bin/cordlr has a merge conflict

File: index.js [hunk 1]
```diff
@@ -1,51 +1,52 @@
...
-            const action = actions.get(name)(bot, options);
-            if (action) action(message, args);
+          const args = spawnargs(raw, { removequotes: 'always' })
+          const flags = minimist(args)
+          const name = args.shift()
+
+          // Run it, if it exists.
+          if (actionNames.indexOf(name) > -1) {
+            const action = actions[name](bot, config)
+            if (action) action(message, args, flags)
```
Comment (AlphaStyle) [inline index.js:37]: the action.js doesn't have the third parameter **`flags`**  yet.
[https://github.com/Devcord/cordlr/blob/master/actions/role.js#L40](url)

Comment (jamen) [reply index.js:37]: Look above.  I added it:

``` js
const flags = minimist(args)
```

It is something optional to use.  `role.js` doesn't use it so we dont have it in the parameters.

Comment (AlphaStyle) [reply index.js:37]: I see! my bad.

**Pull request merged**